src/java/org/apache/cassandra/auth/AuthConfig.java

```diff
@@ -133,4 +124,21 @@ public static void applyAuth()
         cidrAuthorizer.validateConfiguration();
         DatabaseDescriptor.getInternodeAuthenticator().validateConfiguration();
     }
+
+    public static <T> T authInstantiate(ParameterizedClass authCls, Class<T> defaultCls) {
+        if (authCls != null && authCls.class_name != null)
+        {
+            String authPackage = AuthConfig.class.getPackage().getName();
+            return ParameterizedClass.newInstance(authCls, Arrays.asList("", authPackage));
```

Comment:
@tiagomlalves so what happens when autCls is in FQCN form?

Reply:
@smiklosovic thanks for looking at this.

FQCN form is supported by `ParameterizedClass` by specifying an empty search package (first element of the array). This should be required when specifying a own implementation of authentication / authorization classes.

Reply:
@tiagomlalves btw I think that it would be nicer if newInstance just takes `authPackage` and empty string would be covered automatically. `Array.asList` might be also changed to `List.of()`.